org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenImpl.java

```diff
@@ -862,7 +865,17 @@ public boolean isUnavailable(String groupId, String artifactId, String version,
 
     if(artifactFile.canRead()) {
       // artifact is available locally
-      return false;
+      // Need to compare last version to the artifact's one - if they're different then need to re-download
```

Comment:
This seems to be an unrelated optimization; that would better fit in the implementation of Maven itself (and I guess it's already present). Can you please move it to a separate this part in another commit?

Reply:
I don't understand what do you mean by "unrelated optimization" and why should we move it outside of this change...
This check gives a clear signal on that a sources snapshot jar needs to be updated (re-downloaded). Without this change the sources jar, once downloaded will never be updated.
I think making it to be up-to-date (when it's changed on remote repository) is the part of the fix for this exact issue.

Reply:
I mean that this line is in general about skipping update if artifact is already available, and your code is about doing the update whenever there is a newer version available remotely.
This is a good improvement in general, but I don't think it's a requirement for the story of "downloading sources/javadoc with force update".